silver_braintree/payment_processors.py

```diff
@@ -370,9 +373,50 @@ def execute_transaction(self, transaction):
 
         return self._charge_transaction(transaction)
 
+    def find_lost_transaction_id(self, transaction):
```

Comment:
This method is a little bit confusing since it searches for lost transaction and it also takes action in case a missing one is found.

I suggest to split it into a method that only searches and one that reconcile the results.